workflow/cron/operator.go

```diff
@@ -69,16 +81,25 @@ func (woc *cronWfOperationCtx) Run() {
 		return
 	}
 
-	wf := common.ConvertCronWorkflowToWorkflow(woc.cronWf)
+	wf := common.ConvertCronWorkflowToWorkflowWithName(woc.cronWf, getChildWorkflowName(woc.cronWf.Name, scheduledRuntime))
 
 	runWf, err := util.SubmitWorkflow(woc.wfClient, woc.wfClientset, woc.cronWf.Namespace, wf, &v1alpha1.SubmitOpts{})
 	if err != nil {
+		if errors.IsAlreadyExists(err) {
+			// The scheduled workflow already exists, likely indicating that there is a corrupted LastScheduledTime field.
+			// If the intended scheduledRuntime is later than the present value in LastScheduledTime, then replace it
+			if scheduledRuntime.After(woc.cronWf.Status.LastScheduledTime.Time) {
```

Comment:
If it is a conflict error, then our time should be the same surely? This code does not get run?

Reply:
Correct, I think after the overwriting stale `LastScheduledRun` issue is fixed in https://github.com/argoproj/argo/pull/4659 this check will be moot. Not sure I would remove it though as it can only fix a corrupted state if it encounters one. What do you think?

Reply:
I think you need to write tests and then bug fix and maintain all new code - so add less code if it achieves the same outcome

Reply:
Ok, will remove this then